server/src/main/java/org/apache/iotdb/db/query/executor/FillQueryExecutor.java

```diff
@@ -108,6 +113,15 @@ public QueryDataSet execute(QueryContext context)
 
         IFill fill = fillExecutors[i];
 
+        if (fill instanceof LinearFill
```

Comment:
It's better to add test cases to cover unsatisfied data types with `FILL`.

Reply:
I've added tests in last commit